compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -547,159 +519,156 @@ private boolean isPublicInvokableNode(BLangInvokableNode invNode) {
     }
 
     @Override
-    public void visit(BLangBlockFunctionBody body) {
-        boolean prevWithinTxScope = withinTransactionScope;
-        boolean prevLoopAlterNotAllowed = loopAlterNotAllowed;
-        loopAlterNotAllowed = loopCount > 0;
-        if (!transactionalFuncCheckStack.empty() && !withinTransactionScope) {
-            withinTransactionScope = transactionalFuncCheckStack.peek();
-        }
-        final SymbolEnv blockEnv = SymbolEnv.createFuncBodyEnv(body, env);
+    public void visit(BLangBlockFunctionBody body, AnalyzerData data) {
+        boolean prevWithinTxScope = data.withinTransactionScope;
+        boolean prevLoopAlterNotAllowed = data.loopAlterNotAllowed;
+        data.loopAlterNotAllowed = data.loopCount > 0;
+        if (!prevWithinTxScope) {
```

Comment:
There used to be a `!transactionalFuncCheckStack.empty()` check here. Is the removal intentional?

Reply:
Yes, before we visit this method, we push an element to this stack